apps/studio/components/interfaces/TableGridEditor/SidePanelEditor/TableEditor/TableEditor.tsx

```diff
@@ -141,14 +141,18 @@ export const TableEditor = ({
     (constraint) => constraint.type === CONSTRAINT_TYPE.PRIMARY_KEY_CONSTRAINT
   )
 
-  const { data: foreignKeyMeta, isSuccess: isSuccessForeignKeyMeta } =
+  const { data: foreignKeyMeta = [], isSuccess: isSuccessForeignKeyMeta } =
```

Comment:
Nit: Correct me if I'm wrong, but won't `[]` be a new instance every render, so the fallback empty array is never memoized in practice? It doesn't currently cause infinite rerender because of the way we have useEffects set up, but it could be a footgun since at first glance it _looks_ memoized

Reply:
You are right here, let me revert this small change back and post a new PR